test/v2/controllers/V2MovementsControllerSpec.scala

```diff
@@ -0,0 +1,2126 @@
+        ).thenAnswer {
+          invocation =>
+            jsonValidationMockAnswer(MovementType.Arrival)(invocation)
+        }
+
+        when(
+          mockConversionService
+            .jsonToXml(eqTo(MessageType.ArrivalNotification), any[Source[ByteString, _]]())(
+              any[HeaderCarrier],
+              any[ExecutionContext],
+              any[Materializer]
+            )
+        ).thenAnswer {
+          invocation =>
+            EitherT.rightT(
+              invocation.getArgument[Source[ByteString, _]](1)
+            )
+        }
+
+        when(
+          mockMovementsPersistenceService
+            .createMovement(any[String].asInstanceOf[EORINumber], any[MovementType], any[Source[ByteString, _]])(any[HeaderCarrier], any[ExecutionContext])
+        )
+          .thenAnswer(
+            _ =>
+              EitherT.rightT(
+                MovementResponse(MovementId("123"), MessageId("456"))
+              )
+          )
+
+        when(
+          mockRouterService.send(
+            eqTo(MessageType.ArrivalNotification),
+            any[String].asInstanceOf[EORINumber],
+            any[String].asInstanceOf[MovementId],
+            any[String].asInstanceOf[MessageId],
+            any[Source[ByteString, _]]
+          )(any[ExecutionContext], any[HeaderCarrier])
+        )
+          .thenAnswer {
+            _ =>
+              EitherT.leftT(RouterError.UnexpectedError(None))
+          }
+
+        val request = fakeCreateMovementRequest("POST", standardHeaders, singleUseStringSource(CC007Cjson), MovementType.Arrival)
+
+        val result = sut.createMovement(MovementType.Arrival)(request)
+        status(result) mustBe INTERNAL_SERVER_ERROR
+
+        verify(mockValidationService, times(1)).validateJson(eqTo(MessageType.ArrivalNotification), any())(any(), any())
+        verify(mockConversionService).jsonToXml(eqTo(MessageType.ArrivalNotification), any())(any(), any(), any())
+        verify(mockMovementsPersistenceService).createMovement(any[String].asInstanceOf[EORINumber], any[MovementType], any())(any(), any())
+        verify(mockRouterService).send(
+          eqTo(MessageType.ArrivalNotification),
+          any[String].asInstanceOf[EORINumber],
+          any[String].asInstanceOf[MovementId],
+          any[String].asInstanceOf[MessageId],
+          any[Source[ByteString, _]]
+        )(any[ExecutionContext], any[HeaderCarrier])
+      }
+
+    }
+
+    "must return UNSUPPORTED_MEDIA_TYPE when the content type is invalid" in {
+      val standardHeaders = FakeHeaders(
+        Seq(HeaderNames.ACCEPT -> "application/vnd.hmrc.2.0+json", HeaderNames.CONTENT_TYPE -> "invalid", HeaderNames.CONTENT_LENGTH -> "1000")
+      )
+
+      val json    = Json.stringify(Json.obj("CC015" -> Json.obj("SynIdeMES1" -> "UNOC")))
+      val request = fakeCreateMovementRequest("POST", standardHeaders, Source.single(json), MovementType.Arrival)
```

Comment:
same with this class file as well. can we not combine arrival/departure with movemenType list

Reply:
I am doing that for the GET requests, but since the implementation for creating a movement varies for departures and arrivals, I think it makes sense to keep their tests separate